P4Plugin/Source/P4SpecCommand.cpp

```diff
@@ -10,49 +12,106 @@ class P4SpecCommand : public P4Command
 	{
 		ClearStatus();
 		m_Root.clear();
-		
+
+		Conn().Log().Info() << args[0] << "::Run()" << Endl;
+		std::string fallback_error = std::string();
+		if (args.size() > 1)
+			fallback_error = args[1];
+
 		const std::string cmd = std::string("client -o ") + Quote(task.GetP4Client());
-		
+
 		if (!task.CommandRun(cmd, this))
 		{
-			std::string errorMessage = GetStatusMessage();			
-			//This error is handled here instead of in P4LoginCommand as it only shows up when running the first non-login command (currently always spec)
+			//MFA: This error is handled here instead of in P4LoginCommand as it only shows up when running the first non-login command (currently always spec)
 			if (StatusContains(GetStatus(), "login2"))
 			{
 				ClearStatus();
-				GetStatus().insert(VCSStatusItem(VCSSEV_Error, "Your chosen Perforce server has multi-factor authentication enabled. To log in please use a visual client such as p4v or the \"p4 login\" or \"p4 login2\" CLI commands"));
+				std::string exePath = std::string(
+#if defined(_MACOS)
+						"/Applications/HelixMFA.app/Contents/MacOS/HelixMFA"
+#endif
+#if defined(_WINDOWS)
+						"\"C:\\Program Files\\Perforce\\HelixMFA.exe\""
+#endif
+#if defined(_LINUX)
+						"helixmfa" //Just to check if by change someone created a symlink in /usr/local/bin
+#endif
+				);
+				Conn().VerboseLine(exePath);
+				try {
+					POpen proc = POpen(exePath + std::string(" ") + task.GetP4Port() + std::string(" ") + task.GetP4User());
+#if !defined(_WINDOWS)
```

Comment:
ok, as discussed, we now have a better overview;
all this "platform specific" code should be moved in a new POpen::WaitForMFA() function

Reply:
Yes, agreed on accomplishing a deeper refactor asap